linux_container/metrics_test.go

```diff
@@ -241,5 +245,45 @@ system 2
 				})
 			})
 		})
+
+		Describe("network info", func() {
+			It("is returned in the response", func() {
+				metrics, err := container.Metrics()
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(metrics.NetworkStat).To(Equal(garden.ContainerNetworkStat{
+					RxBytes: 1,
+					TxBytes: 2,
+				}))
+			})
+
+			Context("when getting network statistics on non-existent interface", func() {
+				JustBeforeEach(func() {
+					container = linux_container.NewLinuxContainer(
+						lagertest.NewTestLogger("test"),
+						"some-id",
+						"some-handle",
+						containerDir,
+						nil,
+						1*time.Second,
+						containerResources,
+						fake_port_pool.New(1000),
+						fake_command_runner.New(),
+						fakeCgroups,
+						fakeQuotaManager,
+						fake_bandwidth_manager.New(),
+						new(fake_process_tracker.FakeProcessTracker),
+						process.Env{"env1": "env1Value", "env2": "env2Value"},
+						new(networkFakes.FakeFilter),
+						devices.NewLink("non-existent"),
```

Comment:
this should probably use a fake rather than a real link from the device package, container tests should run on any system (so dependencies are normally mocked out) whereas stuff in the devices package is system-specific.